docs/introduction.md

```diff
@@ -0,0 +1,121 @@
+
+## How to use these docs
+
+You can start here and read through these docs linearly like a book; or you can read the specific sections you need. Already familiar with React JS? You can skip that section—or read it for a light refresher.
+
+## Prerequisites
+
+To work with React Native, you will need to have an understanding of JavaScript fundamentals. If you’re new to JavaScript or need a refresher, you can [dive in](https://developer.mozilla.org/en-US/docs/Web/JavaScript) or [brush up](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript) at Mozilla Developer Network.
+
+> While we do our best to assume no prior knowledge of React, iOS, or Android development, these are valuable topics of study for the aspiring React Native developer. Where sensible, we have linked to resources and articles that go more in depth.
+
+## Formatting
+
+Menu paths are written in bold and use carats to navigate submenus. Example: **Android Studio > Preferences**
+
+## Interactive examples
+
+<div class="toggler">
+  <ul role="tablist" id="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTab('syntax', 'functional')">
+      Functional Syntax
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTab('syntax', 'classical')">
+      Class Syntax
+    </li>
+  </ul>
+</div>
+
+<block class="functional webNote iosNote androidNote" />
+
+```SnackPlayer name=Hello%20World
+import React from 'react';
+import { Text, View } from 'react-native';
+
+export default function YourApp() {
+  return (
+      <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
+        <Text>Hello, world!</Text>
+      </View>
+  );
+}
+```
+
+<block class="classical webNote iosNote androidNote" />
+
+```SnackPlayer name=Hello%20World
+import React, { Component } from 'react';
+import { Text, View } from 'react-native';
+
+export default class HelloWorldApp extends Component {
+  render() {
+    return (
+      <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
+        <Text>Hello, world!</Text>
+      </View>
+    );
+  }
+}
+
+```
+
+<block class="classical functional webNote iosNote androidNote" />
+
+The above is a Snack Player. It’s [a handy tool created by Expo](https://expo.io/tools#snack) to embed and run React Native projects on and share how they render in platforms like Android and iOS. We will be using them a lot to demonstrate code samples throughout these docs. You can play around with sample code provided in the Snack, or you can [set up an environment on your local machine](getting-started) and paste the code into your `App.js` file there.
```

Comment:
I think the "call to action" to the docs about setting up your machine should be further up, before the "How to use these docs" section.

Something just like, "It's not necessary, but in case you want to setup your local machine before proceeding you can read the dedicated document". 

I feel that setting up one's machine is one of the most complicated parts of the "learning curve" and having the reference here makes it a bit less visible IMHO

Reply:
Well, that was a bit intentional. Research was showing learners bogging down on environment setup, which is pretty complicated and can be discouraging to folks who just want to get a feel for React Native. The hope is that by moving it past the tutorial mode to a clearly labeled section and using in-browser examples, folks would feel more confident about investing so much time later. But let me have a try!

Reply:
Oh I see!
I think I see your point - and I think it totally makes sense to "move away" the setup from this area.

Reply:
Could we remove the env setup guidance from this paragraph and move the paragraph above this one here (See my comment above)? I believe the flow will work better that way:

* How do use the docs
* You need to know JS
* Play with it right here
* Optionally install a local setup and come back

vs:

* How do use the docs
* You need to know JS
* Optionally install a local setup and maybe come back?
* Play with it right here

The first one gets people to jump in right away, making the local setup a secondary flow for people who just want to get started.